### PR TITLE
Support .NET Framework 4.7.2

### PR DESCRIPTION
**Pull request by rainersigwald**
Based on #2637.

### CODE REVIEW

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR please
Odd hang: https://ci2.dot.net/job/Microsoft_msbuild/job/master/job/innerloop_Windows_NT_CoreCLR_prtest/436/consoleFull#-16758747638101fac1-935e-4556-95f9-7a0e25ca0b52

File: ref/Microsoft.Build.Utilities.Core/net/Microsoft.Build.Utilities.Core.cs [hunk 1]
```diff
@@ -318,6 +318,7 @@ public enum TargetDotNetFrameworkVersion
         Version462 = 10,
         Version47 = 11,
         Version471 = 12,
+        Version472 = 13,
```
Comment (cdmihai) [inline ref/Microsoft.Build.Utilities.Core/net/Microsoft.Build.Utilities.Core.cs]: Shouldn't `VersionLatest` get updated?

Comment (cdmihai) [reply ref/Microsoft.Build.Utilities.Core/net/Microsoft.Build.Utilities.Core.cs]: After reading the comment, maybe bump it to 47 :)

Comment (rainersigwald) [reply ref/Microsoft.Build.Utilities.Core/net/Microsoft.Build.Utilities.Core.cs]: The philosophy we've been going with is "leave it where it is for a full major version", per #1728 and https://github.com/Microsoft/msbuild/pull/2637#discussion_r144913003.

**Pull request merged**